File: Logic & Design/Visual Studio Community Edition/Decision Structures/C++/Roman Numerals/RomanNumerals.cpp
/*
* Programmer:		Corey Jenkins
* Date:				February 8, 2023
* Filename:			RomanNumerals.cpp
* Purpose:			To create a program that gives an error if a Roman Numeral is used outside I through X
*/

// Directives 
#include <iostream>
#include <string>
using namespace std;

void RomanNumerals();

int main()
{
	RomanNumerals();
	system("Pause");
	return 0;
}

void RomanNumerals()
{
	int romanNumeral;

	cout << "Enter a number between 1 to 10: " << endl;
	cin >> romanNumeral;

	switch (romanNumeral)
	{
	case 1:
		cout << "I" << endl;
		break;

	case 2:
		cout << "II" << endl;
		break;

	case 3:
		cout << "III" << endl;
		break;

	case 4:
		cout << "IV" << endl;
		break;

	case 5:
		cout << "V" << endl;
		break;

	case 6:
		cout << "VI" << endl;
		break;

	case 7:
		cout << "VII" << endl;
		break;

	case 8:
		cout << "VIII" << endl;
		break;

	case 9:
		cout << "XI" << endl;
		break;

	case 10:
		cout << "X" << endl;
		break;

	default:
		cout << "An error has occurred." << endl;
		break;
	}

}

File: Logic & Design/Visual Studio Code/Decision Structure and Boolean Logic/Python/roman_numerals.py
'''
Programmer:         Corey Jenkins
Date:               February 8, 2023
Filename:           roman_numerals.py
Purpose:            To create a program that gives an error if a roman numeral is used outside I through X
'''

def main():
    the_roman_numerals()
    
def the_roman_numerals():
    '''
    The pseudocode for this program uses case/switch. So, a while loop will be used for input validation.
    And an if/else statement will be used.
    '''
    
    roman_numeral = int(input("Enter a number between 1 to 10: "))
    while(roman_numeral < 0 or roman_numeral > 10):
        print("ERROR!")
        roman_numeral = int(input("Enter a number between 1 to 10: "))
        
    if(roman_numeral == 1):
        print("I")
    elif(roman_numeral == 2):
        print("II")
    elif(roman_numeral == 3): 
        print("III")
    elif(roman_numeral == 4):
        print("IV")
    elif(roman_numeral == 5):
        print("V")
    elif(roman_numeral == 6):
        print("VI")
    elif(roman_numeral == 7):
        print("VII")
    elif(roman_numeral == 8):
        print("VIII")
    elif(roman_numeral == 9):
        print("IX")
    elif(roman_numeral == 10):
        print("X")
        
main()